05week/spaceTravelToMars.js

```diff
@@ -9,7 +20,48 @@ let jobTypes = {
   programmer: 'Any Ship!'
 };
 
-// Your code here
+//class for crewmembers, crewmembers start with no ship assigned
+class CrewMember {
+  constructor(name, job, specialSkill, ship) {
+    this.name = name;
+    this.job = job;
+    this.specialSkill = specialSkill;
+  }
+
+
+  //crew members can only enter ships based on their job
+  //crew members cannot enter the same ship twice or enter a ship that doesn't exist
+  enterShip(shipName) {
+    console.log(jobTypes[this.job])
+    if((shipName && (shipName.type === jobTypes[this.job] || this.job === 'programmer')) && shipName.crew.indexOf(this) === -1) {
+      this.ship = shipName;
+      shipName.crew.push(this)
+    } else if (!shipName) {
+        console.log(`${shipName} does not exist... yet.`)
+    } else if (shipName.crew.indexOf(this) > -1) {
```

Comment:
This line wouldn't work. You can't check for the index of an object or an array in an array. This would always be false.